### PR TITLE
SCT-624 Store indexing exceptions in DB

### PR DESCRIPTION
**Pull request by MrCreosote**
Sets up the glue code that connects all the changes in previous PRs (e.g. new exception types & error codes, error storage with events) such that indexing errors are now stored in the database along with their associated events.

### CODE REVIEW

File: lib/src/kbasesearchengine/main/IndexerWorker.java [hunk 1]
```diff
@@ -213,25 +213,41 @@ private void logError(
...
         boolean processedEvent = false;
         if (optEvent.isPresent()) {
             final StoredStatusEvent parentEvent = optEvent.get();
             final EventHandler handler;
             try {
                 handler = getEventHandler(parentEvent);
             } catch (UnprocessableEventIndexingException e) {
-                logError(LogPrefix.STD, e);
-                markEventProcessed(parentEvent, StatusEventProcessingState.FAIL);
+                handleException("Error getting event handler", parentEvent, e);
```
Comment (arfathpasha) [inline lib/src/kbasesearchengine/main/IndexerWorker.java]: looks like handleException sets the event state as PROC. Shouldn't it be FAIL in this case?

File: lib/src/kbasesearchengine/main/IndexerWorker.java [hunk 2]
```diff
@@ -213,25 +213,41 @@ private void logError(
...
                 return true;
```
Comment (arfathpasha) [inline lib/src/kbasesearchengine/main/IndexerWorker.java]: Shouldn't this line return false?

File: lib/src/kbasesearchengine/main/IndexerWorker.java (hunk 1 above)
Comment (MrCreosote) [reply lib/src/kbasesearchengine/main/IndexerWorker.java]: No, the PROC tells the method which event state is allowed when looking up the event. It's probably not really needed but doesn't hurt.

Comment (MrCreosote) [reply lib/src/kbasesearchengine/main/IndexerWorker.java]: The method that takes an exception as an argument when setting the state always sets the state to FAIL.

File: lib/src/kbasesearchengine/main/IndexerWorker.java (hunk 2 above)
Comment (MrCreosote) [reply lib/src/kbasesearchengine/main/IndexerWorker.java]: No, the boolean tells the caller whether an event was processed or not, regardless of the final state. The purpose of the boolean is so the calling loop can continue to pull events until there are none left with no delay between events. When this function returns false, the loop will wait.

**Pull request merged**